lnrpc/chainrpc/chain_server.go

```diff
@@ -233,9 +260,51 @@ func (r *ServerShell) CreateSubServer(configRegistry lnrpc.SubServerConfigDispat
 	}
 
 	r.ChainNotifierServer = subServer
+	r.ChainKitServer = subServer
 	return subServer, macPermissions, nil
 }
 
+// GetBlock returns a block given the corresponding block hash.
+func (s *Server) GetBlock(ctx context.Context, in *GetBlockRequest) (
+	*GetBlockResponse, error) {
+
+	// We'll start by reconstructing the RPC request into what the
+	// underlying chain functionality expects.
+	var blockHash chainhash.Hash
+	copy(blockHash[:], in.BlockHash)
+
+	block, err := s.cfg.Chain.GetBlock(&blockHash)
+	if err != nil {
+		return &GetBlockResponse{}, err
+	}
+
+	// Serialize block for RPC response.
+	var blockBuf bytes.Buffer
+	err = block.Serialize(&blockBuf)
+	if err != nil {
+		return &GetBlockResponse{}, err
```

Comment:
I think the gRPC library wants a nil return value of the response message in the error case, otherwise it'll error out itself.

Reply:
How are errors communicated back to the cli?

Reply:
With an error. I mean just this: `return nil, err` instead of `return &GetBlockResponse{}, err`

Reply:
Ah got it, thanks, that makes sense.